proxy_test.go

```diff
@@ -761,3 +762,59 @@ func (r *requestRegistry) get(key string) (bool, error) {
 	}
 	return v, nil
 }
+
+func TestCalcQueryParamsHash(t *testing.T) {
+	testCases := []struct {
+		name           string
+		input          url.Values
+		expectedResult uint32
+	}{
+		{
+			"nil Value",
+			nil,
+			0,
+		},
+		{
+			"empty calcQueryParamsHash",
+			url.Values{},
+			0,
+		},
+		{
+			"map with non param_ value",
+			url.Values{"session_id": {"foo", "bar"}},
+			0,
+		},
+		{
+			"map with only param_ value",
+			url.Values{"param_limit": {"1"}},
+			0x94a386,
+		},
+		{
+			"map with only param_ value. value affects result",
+			url.Values{"param_limit": {"2"}},
+			0x329bae01,
+		},
+		{
+			"map with mix of param_ and non-param_ value",
+			url.Values{"param_limit": {"1"}, "session_id": {"foo", "bar"}},
+			0x94a386,
+		},
+		{
+			"map with multiple param_ values",
+			url.Values{"param_limit": {"1", "2"}, "param_table": {"foo"}},
+			0x3a8a5c31,
+		},
+		{
+			"map with multiple param_ values and only first value in array affects result",
```

Comment:
Why is that? I think I miss some context

Reply:
This is how url.Value works

Reply:
Can it be a real use case? Does Clickhouse accept multiple values for a single query param?